Sources/Datadog/Datadog.swift

```diff
@@ -7,94 +7,152 @@
-    ) {
-        // TODO: RUMM-511 remove this warning
-        #if targetEnvironment(macCatalyst)
-        consolePrint("⚠️ Catalyst is not officially supported by Datadog SDK: some features may NOT be functional!")
-        #endif
-
-        do {
-            try initializeOrThrow(
-                initialTrackingConsent: trackingConsent,
-                configuration: try FeaturesConfiguration(
-                    configuration: configuration,
-                    appContext: appContext
-                ),
-                instanceName: instanceName
-            )
-        } catch {
-            consolePrint("\(error)")
+        /// Either the RUM client token (which supports RUM, Logging and APM) or regular client token, only for Logging and APM.
+        public var clientToken: String
+
+        /// The environment name which will be sent to Datadog. This can be used
+        /// To filter events on different environments (e.g. "staging" or "production").
+        public var env: String
+
+        /// The Datadog server site where data is sent.
+        ///
+        /// Default value is `.us1`.
+        public var site: DatadogSite
+
+        /// The service name associated with data send to Datadog.
+        ///
+        /// Default value is set to application bundle identifier.
+        public var service: String?
+
+        /// The preferred size of batched data uploaded to Datadog servers.
+        /// This value impacts the size and number of requests performed by the SDK.
+        ///
+        /// `.medium` by default.
+        public var batchSize: BatchSize
+
+        /// The preferred frequency of uploading data to Datadog servers.
+        /// This value impacts the frequency of performing network requests by the SDK.
+        ///
+        /// `.average` by default.
+        public var uploadFrequency: UploadFrequency
+
+        /// Proxy configuration attributes.
+        /// This can be used to a enable a custom proxy for uploading tracked data to Datadog's intake.
+        public var proxyConfiguration: [AnyHashable: Any]?
+
+        /// SeData encryption to use for on-disk data persistency by providing an object
+        /// complying with `DataEncryption` protocol.
+        public var encryption: DataEncryption?
+
+        /// A custom NTP synchronization interface.
+        ///
+        /// By default, the Datadog SDK synchronizes with dedicated NTP pools provided by the
+        /// https://www.ntppool.org/ . Using different pools or setting a no-op `ServerDateProvider`
+        /// implementation will result in desynchronization of the SDK instance and the Datadog servers.
+        /// This can lead to significant time shift in RUM sessions or distributed traces.
+        public var serverDateProvider: ServerDateProvider
+
+        /// Sets additional configuration attributes.
+        /// This can be used to tweak internal features of the SDK.
+        public var additionalConfiguration: [String: Any]
```

Comment:
question, blocker-ish/ Should this be user-facing or is it only consumed by cross-platform SDKs? In the latter case, we should move it to `_internal.` scope.

I did some investigation, we introduced it in https://github.com/DataDog/dd-sdk-ios/pull/456 (the _"why?"_ is not well described) as part of [RUMM-1213](https://datadoghq.atlassian.net/browse/RUMM-1213), which belongs to old React-Native SDK epic. I doubt if we want to carry this on in public API - expecting `[String: Any]` map without giving a clue on keys and values doesn't stand for good API design.

Reply:
Agreed, this is used by cross-platforms only therefore it shouldn't exposed here. Thanks for checking!